src/MiniJuvix/Pipeline/EntryPoint.hs

```diff
@@ -2,10 +2,14 @@ module MiniJuvix.Pipeline.EntryPoint where
 
 import MiniJuvix.Prelude
 
+-- | The head of _entryModulePaths is assumed to be the Main module.
 data EntryPoint = EntryPoint
   { _entryRoot :: FilePath,
     _entryModulePaths :: NonEmpty FilePath
   }
   deriving stock (Eq, Show)
 
 makeLenses ''EntryPoint
+
+mainModulePath :: Lens' EntryPoint FilePath
+mainModulePath = entryModulePaths . _head
```

Comment:
Why is this needed?

Reply:
The entry point needs to have a main module. In the monomorphization phase, we use this module to collect the starting set of concrete type applications. As a temporary solution, I pick the first element of the list, which should be the module that the user provided to the CLI. I think it is good to encapsulate this convention in a lens because if we decide to have a specific field for in in the future, we will only need a local update to the definition of this lens